splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/TriggerRowHolderImpl.java

```diff
@@ -0,0 +1,517 @@
+    private boolean             pipelineBufferCreated;
+    private ConglomerateController	cc;
+    private Properties				properties;
+
+    private	ResultDescription		resultDescription;
+    /** Activation object with local state information. */
+    public Activation						activation;
+
+    int 			overflowToConglomThreshold;
+    int 			switchToSparkThreshold;
+    boolean                     isSpark;  // Is the query executing on spark?
+    private ExecRow execRowDefinition;
+    private String  tableVersion;
+
+    protected WriteCoordinator writeCoordinator;
+    protected RecordingCallBuffer<KVPair> triggerTempTableWriteBuffer;
+    protected Callable<Void> triggerTempTableflushCallback;
+    protected Partition triggerTempPartition;
+    private TxnView txn;
+    private byte[] token;
+    private TriggerExecutionContext tec;
+
+    public TriggerRowHolderImpl() {
+
+    }
+
+    /**
+     * Create a temporary row holder with the defined overflow to conglom
+     *
+     * @param activation the activation
+     * @param properties the properties of the original table.  Used
+     *		to help the store use optimal page size, etc.
+     * @param resultDescription the result description.  Relevant for the getResultDescription
+     * 		call on the result set returned by getResultSet.  May be null
+     * @param overflowToConglomThreshold on an attempt to insert
+     * 		this number of rows, the rows will be put
+     *		into a temporary conglomerate.
+     */
+    public TriggerRowHolderImpl
+    (
+            Activation              activation,
+            Properties              properties,
+            ResultDescription       resultDescription,
+            int                     overflowToConglomThreshold,
+            int                     switchToSparkThreshold,
+            ExecRow                 execRowDefinition,
+            String                  tableVersion,
+            boolean                 isSpark,
+            TxnView                 txn,
+            byte[]                  token,
+            long                    ConglomID,
+            TriggerExecutionContext tec
+    )
+    {
+        if (SanityManager.DEBUG)
+        {
+                if (overflowToConglomThreshold < 0)
+                {
+                        SanityManager.THROWASSERT("It is assumed that "+
+                                "the overflow threshold is >= 0.  "+
+                                "If you you need to change this you have to recode some of "+
+                                "this class.");
+                }
+        }
+
+        this.activation = activation;
+        this.properties = properties;
+        this.resultDescription = resultDescription;
+        this.txn = txn;
+        this.token = token;
+        this.tec = tec;
+        this.isSpark = isSpark;
+
+        int initialArraySize = overflowToConglomThreshold < 1 ? 1 : overflowToConglomThreshold;
+        if (initialArraySize > 2000)
```

Comment:
Why do we cap initialArraySize to 2000?

Reply:
To save space.  If we had hundreds of triggers firing simultaneously, each allocated with 1000000 entries initially, even if they are not needed, it would waste a lot of space, which may cause memory pressure.